app/src/main/java/com/brandon/forzenbook/view/composables/CreateUserComposables.kt

```diff
@@ -46,104 +61,211 @@ fun CreateAccountContent(
-            onMaxCharacterLength = { locationError = it },
+    )
+    if (isDialogOpen) {
+        AlertDialog(
+            title = resources.getString(R.string.error_no_internet_connection),
+            body = resources.getString(R.string.error_connect_and_try_again),
+            onDismissRequest = {
+                isDialogOpen = false
+            },
         )
-        if (state.isLocationError) {
-            TextFieldErrorText(text = resources.getString(R.string.createAccountLocationError))
+    }
+    if (datePickerOpen) {
+        LaunchedEffect(Unit) {
+            dateSpinner(context, onDismiss = { datePickerOpen = false }) {
+                dateOfBirth = it
+                datePickerOpen = false
+            }
         }
-        BlackButton(
-            text = resources.getString(R.string.createAccountCreateButtonText),
-        ) {
-            keyboardController?.hide()
-            if (firstName.isNotEmpty() && lastName.isNotEmpty() && dateOfBirth.isNotEmpty() && email.isNotEmpty() && location.isNotEmpty()) {
-                onSubmit(firstName, lastName, dateOfBirth, email, location)
-            } else {
-                emptyFieldError = true
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CreateAccountLoading(
+    state: Loading = Loading(),
+) {
+    val resources = LocalContext.current.resources
+    Scaffold(
+        modifier = Modifier,
+        topBar = { CreateAccountTopBar() },
+        content = { padding ->
+            YellowColumn(
+                modifier = Modifier.padding(padding)
+            ) {
+                InputInfoTextField(
+                    hint = resources.getString(R.string.create_account_first_name_hint),
+                    isEnabled = false,
+                    currentText = state.firstName
+                )
+                InputInfoTextField(
+                    hint = resources.getString(R.string.create_account_last_name_hint),
+                    isEnabled = false,
+                    currentText = state.lastName
+                )
+                ReadOnlyDateTextField(
+                    hint = resources.getString(R.string.create_account_birth_date_format_hint),
+                    isEnabled = false,
+                    currentText = state.dateOfBirth,
+                )
+                InputInfoTextField(
+                    hint = resources.getString(R.string.create_account_email_hint),
+                    isEnabled = false,
+                    currentText = state.email
+                )
+                InputInfoTextField(
+                    hint = resources.getString(R.string.create_account_location_hint),
+                    isEnabled = false,
+                    currentText = state.location
+                )
+                LoadingBlackButton()
             }
         }
-        if (emptyFieldError) {
-            TextFieldErrorText(text = resources.getString(R.string.requiredFieldsError))
+    )
+}
+
+fun dateSpinner(
```

Comment:
Might be more appropriate to call this "dateDialog"

Reply:
renamed to dateDialog